### ISSUE TITLE
API documentation is woeful

### ISSUE THREAD

**Issue by jmm** [google/traceur-compiler#1668]
It's difficult to figure out how to even get off the ground with the API. [This document](https://github.com/google/traceur-compiler/wiki/Node.js-API) isn't much help.

What is the `1.0.0` it refers to?

There don't  seem to be even "hello world" level examples for common use cases like following the dependency graph of 1 or more entry files. And the source code is far from easy to follow to glean this information (holy cow that `bin/traceur.js` file is a monster). From poking around in the source I've gotten as far as figuring out how to get the ball rolling with something like this:

```js
var opts = {
  modules: 'commonjs',
  sourceMaps: 'inline',
};

traceur.recursiveModuleCompileToSingleFile('./output', [{name: './input'}], opts);
```

And then saw [this example](https://github.com/google/traceur-compiler/issues/1626#issuecomment-69777171):

```js
var traceurAPI = require('./api.js');
var out = './output.js';
var sources =  {name: 'input.js', type: 'module'};
var options = {};
traceurAPI.recursiveModuleCompileToSingleFile(out, sources, options).then(function() {
      process.exit(0);
    }).catch(function(err) {
      var errors = err.errors || [err];
      errors.forEach(function(err) {
        console.error(err.stack || err);
      });
      process.exit(1);
    });
```

With some corrections and comments, ts that example suitable to add to the [Node.js API](https://github.com/google/traceur-compiler/wiki/Node.js-API) page as a "hello world" level example for that use case? Here are a couple of issues I've noticed with it so far:

* I don't know why it's `var traceurAPI = require('./api.js');` -- should be `require('traceur')` I'd think.

* It seems that the `sources` argument must be an array.

**Comment by UltCombo** [comment]
If you want to run non-transpiled ES.next code in Node.js with Traceur runtime compilation then you can use [`traceur.require.makeDefault([filter, options])`](https://github.com/google/traceur-compiler/wiki/Using-Traceur-with-Node.js).

**Comment by jmm** [comment]
@UltCombo Thanks, I'm talking about using the API to compile code though.

**Comment by johnjbarton** [comment]
I fixed the page https://github.com/google/traceur-compiler/wiki/Node.js-API. 

`bin/traceur.js` is, like the name implies, a binary-like files. Don't read it.

The fundamental problem with our API (and why our documentation is indeed woeful as you say) is simply that we don't agree on what the use cases should be.  As a practical matter several related projects have created covering APIs based on their favorite use case.

**Comment by UltCombo** [comment]
Oh my bad, I skimmed through your code and didn't see the `out` parameter.

Perhaps taking a look at how [grunt-traceur](https://github.com/aaronfrost/grunt-traceur) and [gulp-traceur](https://github.com/sindresorhus/gulp-traceur) consume the API would be of use?

And indeed, there are quite a few API functions that are not documented (see [`src/node/api.js`](https://github.com/google/traceur-compiler/blob/master/src/node/api.js)'s exports).

**Comment by jmm** [comment]
@UltCombo 
> Perhaps taking a look at how grunt-traceur and gulp-traceur consume the API would be of use?

Perhaps. It somewhat depends on whether they're utilizing parts of the API I'm interested in. What got me started on this today is figuring out why with browserify and es6ify I was getting local paths that I don't want exposed in my bundle as `sourceURL` values. I ultimately did end up looking at what traceur API calls es6ify is making as part of my debugging.

Looking at `src/node/api.js` (in the NPM package) is how I got to `recursiveModuleCompileToSingleFile()`, but I also pretty quickly got in some pretty deep weeds.

**Comment by jmm** [comment]
@johnjbarton So is that the only compilation-related API that can be documented right now?

> `bin/traceur.js` is, like the name implies, a binary-like files. Don't read it.

I was working from the npm package. I guess I need to look in the repo if I'm going to have to dig around like this.

> As a practical matter several related projects have created covering APIs based on their favorite use case.

Are you referring to projects like es6ify?

> I fixed the page https://github.com/google/traceur-compiler/wiki/Node.js-API.

Instead of:

```js
var api = require('../../../src/node/api');
```

Shouldn't it be?:

```js
var api = require('traceur');
```

**Comment by johnjbarton** [comment]
> Are you referring to projects like es6ify?
Yes, and gulp-traceur, its grunt cousin, es-module-loader, and its cousin jspm and so on.

I updated the wiki page again. I'll add a test to our repo to verify it later today.

**Comment by jmm** [comment]
> Yes, and gulp-traceur, its grunt cousin, es-module-loader, and its cousin jspm and so on.

Ok, thanks.

> I updated the wiki page again. I'll add a test to our repo to verify it later today.

I see it's now `require('traceur/src/node/api.js')`. Isn't `require('traceur')` equivalent because of the value of `package.json` `main`? Is the expectation that `main` is going to change?
